jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/ThriftSpanConverter.java

```diff
@@ -76,7 +77,32 @@
     List<LogData> logs = span.getLogs();
     if (logs != null) {
       for (LogData logData : logs) {
-        annotations.add(new Annotation(logData.getTime(), logData.getMessage()));
+        // skip if log is empty
+        if (logData.getMessage() == null && logData.getPayload() == null &&
+                (logData.getFields() == null || logData.getFields().isEmpty())) {
+          continue;
+        }
+
+        String value = logData.getMessage();
+        if (logData.getPayload() != null) {
+          value = logData.getPayload().toString();
+        } else if (logData.getFields() != null && !logData.getFields().isEmpty()) {
+          // special-case the "event" field which is similar to the semantics of a zipkin annotation
+          Object event = logData.getFields().get("event");
+          if (event != null && logData.getFields().size() == 1) {
+            value  = event.toString();
+          } else {
+            StringBuilder result = new StringBuilder();
+            for (Iterator<? extends Map.Entry<String, ?>> i = logData.getFields().entrySet().iterator(); i.hasNext(); ) {
```

Comment:
might want to comment that this is the existing practice in zipkin-go-opentracing.. we do need to put together an OT mapping page on the website.. something bas and I spoke of in the past.